File: android/arsenal/libraries/library-base/src/main/java/org/smartrobot/database/model/OrderPayStatusConverter.java
package org.smartrobot.database.model;


import org.greenrobot.greendao.converter.PropertyConverter;

public class OrderPayStatusConverter implements PropertyConverter<OrderPayStatus, String> {

    @Override
    public OrderPayStatus convertToEntityProperty(String databaseValue) {
        return OrderPayStatus.valueOf(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(OrderPayStatus entityProperty) {
        return entityProperty.name();
    }
}

File: android/arsenal/libraries/library-base/src/main/java/org/smartrobot/database/model/OrderPayStatusConverter.kt
package org.smartrobot.database.model

import org.greenrobot.greendao.converter.PropertyConverter

class OrderPayStatusConverter : PropertyConverter<OrderPayStatus, String> {
    override fun convertToEntityProperty(databaseValue: String): OrderPayStatus {
        return OrderPayStatus.valueOf(databaseValue)
    }

    override fun convertToDatabaseValue(entityProperty: OrderPayStatus): String {
        return entityProperty.name
    }
}